test/libp2p/protocol/gossip/gossip_structures_test.cpp

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <libp2p/protocol/gossip/impl/gossip_wire_protocol.hpp>
+#include <libp2p/protocol/gossip/impl/message_cache.hpp>
+#include <libp2p/protocol/gossip/impl/peers.hpp>
+
+#include <gtest/gtest.h>
+#include <spdlog/fmt/fmt.h>
+#include "testutil/libp2p/peer.hpp"
+
+static bool VERBOSE = false;
+
+using fmt::print;
+
+#define TR(var) \
+  if (VERBOSE)  \
+  fmt::print("{}: {}={}\n", __LINE__, #var, var)
+
+namespace g = libp2p::protocol::gossip;
+
```

Comment:
Tests are hard to read and without comments, thus they are unmaintainable. Common approach in our projects is to have tests structured with `given when then`. Take a look on other tests for example and rewrite